model/SearchEngine/Driver/Elasticsearch/ElasticSearchConfig.php

```diff
@@ -26,7 +26,7 @@
 
 use oat\generis\model\DependencyInjection\ServiceOptionsInterface;
 
-class ElasticSearchConfig
+class ElasticSearchConfig implements ElasticSearchConfigInterface
```

Comment:
I would honestly avoid creating factories, new entities, etc and just fallback to config file info in case the ENV vars do not exist. There is no need to complicate is if we plan to use ENV vars fallback. I think it is too much of effort in this moment and creating more classes to deprecate

Reply:
The main reason for splitting between env and simple config is to avoid having a lot of `if` inside `ElasticSearchConfig` which make it not readable. For sure, it will never be updated in the future if we will decide to use always env variables.

But ok, I don't want to spend a lot of time on that and will rollback all my changes just to speed up it

Reply:
I understood the logic, but since we are moving towards config.php or env var, I think it is to soon to introduce drivers for config. 

We did introduce it for some apps like TEMPO, etc and it actually increased the maintenance cost for us as having to update several drivers every time. So for this config case, I am not sure it worth it. 

It is my personal opinion, so take it lightly and revert if you agree. :)